File: goormlevel/195693/1. 통증-2/통증 2.py
import sys

input = sys.stdin.readline
write = sys.stdout.write

def solve():
    n = int(input().rstrip())
    a, b = map(int, input().split())

    bval = n // b

    while bval >= 0:
        remains = n - (bval * b)
        
        if remains % a == 0:
            aval = remains // a
            write(str(bval + aval) + "\n")
            return
        
        bval -= 1
        
    write("-1\n")

if __name__ == "__main__":
    solve()

File: goormlevel/195693/1. 통증-2/통증 2.c
#include <stdio.h>

int main() {
    int n, a, b;
    
    if (scanf("%d %d %d", &n, &a, &b) != 3) {
        return 0;
    }

    int bval = n / b;

    while (bval >= 0) {
        int remains = n - (bval * b);
        
        if (remains % a == 0) {
            int aval = remains / a;
            printf("%d\n", bval + aval);
            return 0;
        }
        
        bval--;
    }
    
    printf("-1\n");
    
    return 0;
}
